Standardize rustdoc subheadings

jamillambert opened the pull request:
The subheadings in the rustdocs have been standardized according to [./CONTRIBUTING.md](https://github.com/rust-bitcoin/rust-bitcoin/blob/master/CONTRIBUTING.md):
```rust
impl FooBar {
    /// Constructs a `FooBar` from a [`Baz`].
    ///
    /// # Errors
    ///
    /// Returns an error if `Baz` is not ...
    ///
    /// # Panics
    ///
    /// If the `Baz`, converted to a `usize`, is out of bounds.
    pub fn from_baz(baz: Baz) -> Result<Self, Error> {
        ...
    }
}
```

apoelstra commented on bitcoin/src/taproot/mod.rs:
```diff
@@ -429,8 +429,12 @@ impl TaprootBuilder {
+    /// The depth of the root node is 0.
+    ///
+    /// # Errors
+    ///
```
In f52afae4fcfdd9ab013cb56bc297c7eef9c73baf:

trailing whitespace here

jamillambert replied:
Removed the white space below on line 454

apoelstra commented on units/src/amount.rs:
```diff
@@ -1046,12 +1046,20 @@ impl Amount {
+    ///
+    /// # Panics
+    ///
+    /// Panics in debug mode, wraps in release mode.
```
In 3f76015ee911a78c9e7cb65953166dad2614cda6:

Can you clarify this to say "**On overflow,** panics in debug mode and wraps in release mode.".

jamillambert replied:
Clarified in both `unchecked_add()` and `unchecked_sub()`

tcharding reviewed (commented):
Oh, you fixed review suggestions as a separate patch. My review highlights why that is not the way we do it here (some projects do).

If you put an additional patch then next reviewer (in this case me), ends up picking up the same mistakes as the first reviewer because I pull down the branch and review it patch by patch.

So in future, when you are fixing review suggestions please use `git rebase` and amend the patch that you are changing. Then do a `git push -f` to force push the new version. FTR reviewers that acked already will then use `git range-diff` to check what changed in the force push.

Remember to read all your own patches after rebasing to make sure changes are in the right patch :)

You might as well practice on this one.

tcharding commented on bitcoin/src/taproot/mod.rs:
```diff
@@ -441,16 +445,24 @@ impl TaprootBuilder {
     /// See [`TaprootBuilder::add_leaf_with_ver`] for adding a leaf with specific version.
+    ///
+    /// # Errors
+    ///
```
This is fixed in edfb18aa0bdb9208be035f3ee02fbe61a90456fe

jamillambert replied:
Done, and thanks for the shell function.

jamillambert commented on bitcoin/src/taproot/mod.rs:
```diff
@@ -429,8 +429,12 @@ impl TaprootBuilder {
-    /// Adds a leaf script at `depth` to the builder with script version `ver`. Errors if the leaves
-    /// are not provided in DFS walk order. The depth of the root node is 0.
+    /// Adds a leaf script at `depth` to the builder with script version `ver`.
+    /// The depth of the root node is 0.
```
Done

jamillambert commented on bitcoin/src/taproot/mod.rs:
```diff
@@ -441,16 +445,24 @@ impl TaprootBuilder {
-    /// Adds a hidden/omitted node at `depth` to the builder. Errors if the leaves are not provided
-    /// in DFS walk order. The depth of the root node is 0.
+    /// Adds a hidden/omitted node at `depth` to the builder.
+    /// The depth of the root node is 0.
```
done

jamillambert commented:
Thanks for your advice.  I have removed the patch with the review suggestions and incorporated them in the original patches.  It does make more sense to do it this way.

apoelstra reviewed (approved):
ACK 11bb1ff6ff65e89d9f101ffb7d76ebc09a20d738

tcharding reviewed (approved):
ACK 11bb1ff6ff65e89d9f101ffb7d76ebc09a20d738

tcharding commented:
Thanks bro.

tcharding commented:
Hey @jamillambert you could fix all the warnings under a standard heading too if you feel like it. If you use `git grep -i warning` there are a bunch of non-standard things.

ref: https://github.com/rust-bitcoin/rust-bitcoin/pull/2795#discussion_r1613023238
